app/src/main/java/at/bitfire/icsdroid/Settings.kt

```diff
@@ -6,44 +6,37 @@ package at.bitfire.icsdroid
 
 import android.content.Context
 import android.content.SharedPreferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.longPreferencesKey
 import androidx.lifecycle.LiveData
 import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.single
+import kotlinx.coroutines.runBlocking
 
 class Settings(context: Context) {
 
     companion object {
-        private const val FORCE_DARK_MODE = "forceDarkMode"
-    }
-
+        @Deprecated("Use DataStore")
+        const val FORCE_DARK_MODE = "forceDarkMode"
 
-    private val prefs: SharedPreferences = context.getSharedPreferences("icsx5", 0)
+        val ForceDarkMode = booleanPreferencesKey("forceDarkMode")
```

Comment:
Any specific reason the variable name is capitalized?

Reply:
It's technically written as a constant in [the docs](https://developer.android.com/topic/libraries/architecture/datastore#preferences-read) (all caps). However, it's technically not a constant, and that's why I wrote it in camel-case capital-first, but I agree that it should be lowercase.